nexus/reconfigurator/planning/src/blueprint_builder/builder.rs

```diff
@@ -1890,18 +1844,179 @@ impl<'a> BlueprintBuilder<'a> {
     /// ordinarily only come from RSS.
     ///
     /// TODO-cleanup: Remove when external DNS addresses are in the policy.
-    pub(crate) fn add_external_dns_ip(
+    // This can't be `#[cfg(test)]` because it's used by the `ExampleSystem`
+    // helper (which itself is used by reconfigurator-cli and friends). We give
+    // it a scary name instead.
+    pub(crate) fn inject_untracked_external_dns_ip(
         &mut self,
         addr: IpAddr,
     ) -> Result<(), Error> {
-        self.external_networking()?.add_external_dns_ip(addr)
+        Ok(self.resource_allocator()?.inject_untracked_external_dns_ip(addr)?)
+    }
+}
+
+// Helper to validate that the system hasn't gone off the rails. There should
```

Comment:
This function moved wholesale from `external_networking.rs` without any changes. I noted where we call it above, but I think ultimately this should move to `blippy`.